src/payment/strategies/digitalriver/digitalriver-payment-strategy.ts

```diff
@@ -0,0 +1,167 @@
+            },
+            onReady: (data?: OnReadyResponse) => {
+                this._onReadyResponse(data);
+            },
+            onError: (error: OnCancelOrErrorResponse) => {
+                this._getDigitalRiverInitializeOptions().onError?.(new Error(this._getErrorMessage(error)));
+            },
+        };
+
+        this._digitalRiverJS = await this._digitalRiverScriptLoader.load(paymentMethod.initializationData.publicKey, paymentMethod.initializationData.paymentLanguage);
+        this._digitalRiverDropComponent = await this._getDigitalRiverJs().createDropin( configuration );
+        await this._digitalRiverDropComponent.mount(this._getDigitalRiverInitializeOptions().containerId);
+
+        return state;
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = orderRequest;
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+        if (!payment || !this._loadSuccessResponse || !this._digitalRiverCheckoutId) {
+            throw new InvalidArgumentError('Unable to proceed because payload payment argument is not provided.');
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    private _getDigitalRiverJs(): DigitalRiverJS {
+        if (!this._digitalRiverJS) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._digitalRiverJS;
+    }
+
+    private _getErrorMessage(error: OnCancelOrErrorResponse): string {
+        const { errors } = error;
+
+        return errors.map(e => 'code: ' + e.code + ' message: ' + e.message).join('\n');
+    }
+
+    private _onSuccessResponse(data?: OnSuccessResponse): Promise<void> {
+        return new Promise((resolve, reject) => {
+            if (data && this._submitFormEvent) {
+                const { browserInfo } = data.source;
+                this._loadSuccessResponse = browserInfo ? {
+                    source: {
+                        id: data.source.id,
+                        reusable: data.source.reusable,
+                        ...browserInfo,
+                    },
+                    readyForStorage: data.readyForStorage,
+                } : {
+                    source: {
+                        id: data.source.id,
+                        reusable: data.source.reusable,
+                    },
+                    readyForStorage: data.readyForStorage,
+                } ;
+                resolve(this._submitFormEvent());
+            }
+
+            reject(this._getDigitalRiverInitializeOptions().onError?.
```

Comment:
you are calling `reject()` with a function that doesn't return anything, instead you should probably have
```js
const error = new InvalidArgumentError('Unable to initialize payment because success argument is not provided.')
reject(error);
this._getDigitalRiverInitializeOptions().onError?(error)